Let's try this again

lucywyman opened the pull request:
I'm not sure whether this aligns with your vision for the project, but I added everything to it's own file, so you can integrate as desired. Let me know if you want more help with it!

lucywyman committed:
Let's try this again

ElijahCaine commented:
Could you describe the changes you made? I'm not sure what I'm looking at. 

(also its README.rst not README.md, just a minor tweak)

lucywyman commented:
* Included manifest file to specify default files in distributed package
* Added copy_file and copy_dir to, well, copy files and directories from a source to destination
* Modified grab to download files to a data directory created in the users current directory (TODO allow them to specify a directory)
* Modified build to write output to it's own directory
* from os import ... -> import os (I think I tried to use path as a variable, then ended up not doing that because it was ambiguous, though I think this is still good practice)

ElijahCaine commented:
@lucywyman I appreciate the PR, unfortunately I don't really think it aligns with the design principles of ``pageup``.

* The build and data dirs are a good idea (I mean... they're used in pretty much every popular SSG I can think of) but pageup is meant to be a lot more simple. If you're making a project that requires a ``build`` dir, pageup is probably not what you're looking for. It also muddies up the URL schema I have (with a build dir either the user has to move the files after building or the url looks like ``example.com/mypage/build/`` instead of just ``example.com/mypage/``)

* The use of urllib is fine as it is cleaner code than using requests, but we actually migrated away from requests early on as it was suggested by some peers that it's not a good library for downloading files. Granted, the user needs to download requests to make pageup go, but they also need to download jinja2 and markupsafe so might as well throw in another useful lib while we're at it. (If you're gonna do it, do it right).
